server/lib/pbench/process_tb.py

```diff
@@ -0,0 +1,117 @@
+import os
+import subprocess
+
+from collections import namedtuple
+from logging import Logger
+from pathlib import Path
+
+from pbench import PbenchConfig
+
+Results = namedtuple("Results", ["nstatus", "ntotal", "ntbs"])
+
+
+class ProcessTb:
+    def __init__(self, config: PbenchConfig, logger: Logger):
+        """ Processes Tar ball, gets the tar ball from receiving directory
+        and copies it to the remote server by creating a subprocess which
+        call Agent's 'pbench-results-push' command.
+
+            Args -
+                config -- PbenchServer config object
+                logger -- logger object to use when emitting log entries during
+                            operation
+
+        """
+        self.config = config
+        self.logger = logger
+        self.receive_dir = self.get_receive_dir()
+
+    def get_receive_dir(self) -> Path:
+        receive_dir_prefix = self.config.get(
+            "pbench-server", "pbench-receive-dir-prefix"
+        )
+        if not receive_dir_prefix:
+            raise ValueError(
+                "Failed: No value for config option pbench-receive-dir-prefix in section pbench-server"
+            )
+
+        receive_dir = Path(f"{receive_dir_prefix}-002").resolve(strict=True)
+        if not receive_dir.is_dir():
+            raise NotADirectoryError(
+                "Failed: {} does not exist, or is not a directory", receive_dir
+            )
+
+        return receive_dir
+
+    def run_subprocess(self, controller: str, tb: Path, token: str):
+        """Runs `pbench-results-push` command with controller, tb
+        and token options
+
+        Args -
+            controller -- the name of the controller to be associated with the
+                          tar ball
+            tb -- path of the tar ball
+            token -- generated authorised token for Pbench user
+
+        Importance of this function is while running tests we get the
+        ability to mock this function and test it easily
+        """
+
+        res = subprocess.run(
+            ["pbench-results-push", controller, tb, token], capture_output=True
+        )
+        error = res.stderr
+        if res.returncode == 1:
+            raise RuntimeError(error)
+
+    def process_tb(self) -> Results:
+        """Receives tar ball and copies it to the remote server
+            using Agent's 'pbench-results-push' command
```

Comment:
Searches for tarballs in the configured receive directory and uploads each using the Agent pbench-results-push? (Although even that is indirect, as "uses pbench-results-push" is really controlled by the `run_subprocess` method, not here, and it's better to document what *this* method is doing, not what other methods were expected to do at the time this method was documented.)